truss-chains/truss_chains/deploy.py

```diff
@@ -338,71 +328,423 @@ def deploy_remotely(
+        ), "If the chain is draft, the oracles must be draft."
+
+        self._chainlet_data = _map_chainlet_data(deployed_chainlets)
+        self._assert_chainlet_names_same(chainlet_names)
+        self._ignore_patterns = truss_path.load_trussignore_patterns()
+
+        def watch_filter(_, path):
+            return not truss_path.is_ignored(pathlib.Path(path), self._ignore_patterns)
+
+        logging.getLogger("watchfiles.main").disabled = True
+        self._watch_filter = watch_filter
 
-    return chain_service
+    @property
+    def _original_chainlet_names(self) -> set[str]:
+        return set(self._chainlet_data.keys())
+
+    @property
+    def _chainlet_display_name_to_url(self) -> Mapping[str, str]:
+        return {k: v.oracle_predict_url for k, v in self._chainlet_data.items()}
+
+    def _assert_chainlet_names_same(self, new_names: set[str]) -> None:
+        missing = self._original_chainlet_names - new_names
+        added = new_names - self._original_chainlet_names
+        if not (missing or added):
+            return
+        msg_parts = [
+            "The deployed Chainlets and the Chainlets in the current workspace differ. "
+            "Live patching is not possible if the set of Chainlet names differ."
+        ]
+        if missing:
+            msg_parts.append(f"Chainlets missing in current workspace: {list(missing)}")
+        if added:
+            msg_parts.append(f"Chainlets added in current workspace: {list(added)}")
+
+        raise definitions.ChainsDeploymentError("\n".join(msg_parts))
+
+    def _code_gen_and_patch_thread(
+        self, descr: definitions.ChainletAPIDescriptor, user_env: Mapping[str, str]
+    ) -> tuple[b10_remote.PatchResult, list[str]]:
+        with log_utils.LogInterceptor() as log_interceptor:
+            # TODO: Maybe try-except code_gen errors explicitly.
+            chainlet_dir = code_gen.gen_truss_chainlet(
+                self._chain_root,
+                pathlib.Path(tempfile.gettempdir()),
+                self._deployed_chain_name,
+                descr,
+                self._chainlet_data[descr.display_name].oracle_name,
+                self._chainlet_display_name_to_url,
+                user_env,
+            )
+            patch_result = self._remote_provider.patch_for_chainlet(
+                chainlet_dir, self._ignore_patterns
+            )
+            logs = log_interceptor.get_logs()
+        return patch_result, logs
+
+    def _patch(
+        self,
+        executor: concurrent.futures.Executor,
+        user_env: Optional[Mapping[str, str]],
+    ) -> None:
+        exception_raised = None
+        with log_utils.LogInterceptor() as log_interceptor, self._console.status(
+            " Live Patching Chain.\n", spinner="arrow3"
+        ):
+            # Handle import errors gracefully (e.g. if user saved file, but there
+            # are syntax errors, undefined symbols etc.).
+            try:
+                with framework.import_target(
```

Comment:
could you leave a comment explaining why we need to do this?

Reply:
Is this question to why it's in a `try`-block, or why the target is imported?

We need to import it to get the initial chain name and set of chainlets and such things.